internal/app/machined/pkg/controllers/cluster/local_affiliate.go

```diff
@@ -232,6 +242,28 @@ func (ctrl *LocalAffiliateController) Run(ctx context.Context, r controller.Runt
 							endpoints = append(endpoints, netip.AddrPortFrom(ip, constants.KubeSpanDefaultPort))
 						}
 
+						// mix in discovered public IPs
+						for iter := safe.IteratorFromList(discoveredPublicIPs); iter.Next(); {
+							addr := iter.Value().TypedSpec().Address.Addr()
+
+							duplicate := false
+
+							for _, ip := range nodeIPs {
+								if ip == addr {
+									duplicate = true
+
+									break
+								}
+							}
+
+							if duplicate {
+								// this address already found
+								continue
+							}
```

Comment:
```suggestion
							if slices.Index(nodeIPs, addr) != -1 {
								// this address already found
								continue
							}
```

Reply:
ha, it's not actually available in our `gen` package